projects/igniteui-angular/schematics/ng-add/index.ts

```diff
@@ -145,6 +167,47 @@ function addPackageToJsonDevDependency(tree: Tree, pkg: string, version: string)
   return tree;
 }
 
+function displayVersionMismatch(context: SchematicContext, tree: Tree, igPackageJson: any) {
+  const ngKey = '@angular/core';
+  const ngCliKey = '@angular/cli';
+  const ngProjVer = getDependencyVersion(ngKey, tree);
+  const ngCliProjVer = getDependencyVersion(ngCliKey, tree);
+
+  const igAngularVer = igPackageJson.peerDependencies[ngKey];
+  const igAngularCliVer = igPackageJson.peerDependencies[ngCliKey];
+
+  if (ngProjVer < igAngularVer) {
+    context.logger.log('warn',
+      `Angular version mismatch. Installed version: ${ngProjVer} - igniteui-angular version: ${igAngularVer}`);
+  }
+  if (ngCliProjVer < igAngularCliVer) {
+    context.logger.log('warn',
+      `Angular CLI version mismatch. Installed version: ${ngCliProjVer} - igniteui-angular version: ${igAngularCliVer}`);
```

Comment:
I'd check `@angular/common` instead of CLI.
The message could be a bit more descriptive too. Something explaining the added Ignite UI for Angular package is build against a newer version of angular[/core] and lower versions were found in the current project.
Could also suggest a way to remedy that would be to use `ng update`